### DISCUSSION TITLE
Parameters not typed

### DISCUSSION THREAD

**Discussion by kristiandupont** [trpc/trpc#1450] [Q&A]
I'm trying to set tRPC up and I have created the "hello" sample from the documentation. In the client, it correctly suggests `"hello"` as the only possible path name, but the second parameter is just `...args: any[]`, and predictably, I get no compiler errors when I give something wrong. What could be the issue here?

<img width="695" alt="image" src="https://user-images.githubusercontent.com/847742/150818275-f3b111f5-b652-4e5a-a4f9-e93eb417a92a.png">

**Comment by KATT** [comment]
I need access repo to know what can cause it but

- compare your tsconfig to https://github.com/trpc/trpc/blob/1fbbe80c79f7a3ad7c52e7be793342537dcbecbd/examples/next-prisma-starter/tsconfig.json#L11-L12
- check that you don't have any type errors in your setup files
- compare each file according [to the example closest to your usecase](https://github.com/trpc/trpc/tree/main/examples)

**Reply by kristiandupont** [reply]
So it seems that the problem is related to a missing `@trpc/server` in my client package.json. Am I supposed to add it to my client as a dev-dependency? If I do that, I get a conflict with a `middlewareMarker` somewhere, which is a symbol and thus doesn't match across the two projects.

**Reply by KATT** [reply]
Try `yarn add @trpc/server` then

**Reply by kristiandupont** [reply]
It gives me this:
```
src/trpc.ts:7:40 - error TS2344: Type 'Router<unknown, unknown, Record<"hello", Procedure<unknown, unknown, { text?: string | null | undefined; } | null | undefined, { text?: string | null | undefined; } | null | undefined, { greeting: string; }>>, {}, {}, DefaultErrorShape>' does not satisfy the constraint 'Router<any, any, any, any, any, any>'.
  The types of '_def.middlewares' are incompatible between these types.
    Type 'MiddlewareFunction<unknown, unknown>[]' is not assignable to type 'MiddlewareFunction<any, any>[]'.
      Type 'MiddlewareFunction<unknown, unknown>' is not assignable to type 'MiddlewareFunction<any, any>'.
        Types of parameters 'opts' and 'opts' are incompatible.
          Type '{ ctx: any; type: ProcedureType; path: string; rawInput: unknown; next: { (): Promise<MiddlewareResult<any>>; <T>(opts: { ctx: T; }): Promise<MiddlewareResult<T>>; }; }' is not assignable to type '{ ctx: unknown; type: ProcedureType; path: string; rawInput: unknown; next: { (): Promise<MiddlewareResult<unknown>>; <T>(opts: { ctx: T; }): Promise<MiddlewareResult<T>>; }; }'.
            The types returned by 'next()' are incompatible between these types.
              Type 'Promise<MiddlewareResult<any>>' is not assignable to type 'Promise<MiddlewareResult<unknown>>'.
                Type 'MiddlewareResult<any>' is not assignable to type 'MiddlewareResult<unknown>'.
                  Type 'MiddlewareOKResult<any>' is not assignable to type 'MiddlewareResult<unknown>'.
                    Type 'MiddlewareOKResult<any>' is not assignable to type 'MiddlewareOKResult<unknown>'.
                      Types of property 'marker' are incompatible.
                        Type 'typeof import("/usr/workspace/repositories/mymee/web-apps/node_modules/@trpc/server/dist/declarations/src/internals/middlewares").middlewareMarker' is not assignable to type 'typeof import("/usr/workspace/repositories/mymee/backend/node_modules/@trpc/server/dist/declarations/src/internals/middlewares").middlewareMarker'.
```

Basically, since the `middlewareMarker` is a `symbol`, having one in my `backend/node_modules/...` and another one in `frontend/node_modules/...` causes this conflict.

I know the docs say that this is meant for monorepo projects -- is it also meant for mono-`package.json` projects?

**Reply by KATT** [reply]
I don't have access to your repo, I can't help you. The best thing I can do is to ask you to look at the examples that are working perfectly fine.

- Single app: https://github.com/trpc/examples-next-prisma-starter
- Monorepo: https://github.com/katt/zart

Maybe start from one of the example projects.

**Reply by kristiandupont** [reply]
That's ok, I'll investigave further and update this if I figure it out, just in case someone else runs into a similar problem. Thank you for your help and for creating the library!

**Reply by KATT** [reply]
Thank you!

**Reply by kristiandupont** [reply]
So just a quick update: it turned out that I had installed alpha versions of trpc in my project. I am not sure how that happened, but reverting to 9.17.1 seems to have solved the problem.

**Reply by KATT** [reply]
Ohhh shit, I accidentally published those as `latest` for ~10 hours a few days ago and reverted when someone noticed, I didn't think of that!

**Reply by KATT** [reply]
Thanks for getting back with the solution :)
